src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseImportFromFileService.java

```diff
@@ -131,6 +131,9 @@ private void copyImportedExerciseContentToRepositories(Repository templateRepo,
     private void copyExerciseContentToRepository(Repository repository, RepositoryType repositoryType, Path basePath) throws IOException {
         FileUtils.copyDirectory(retrieveRepositoryDirectoryPath(basePath, repositoryType.getName()).toFile(), repository.getLocalPath().toFile(),
                 new NotFileFilter(new NameFileFilter(".git")));
+        try (var files = Files.walk(repository.getLocalPath())) {
+            files.filter(file -> file.getFileName().endsWith("gradlew")).forEach(file -> file.toFile().setExecutable(true));
```

Comment:
So if I name my file `helloWorldMygradlew`, it becomes executable? I think that's a bit weird... Can you somehow make it so that it fully checks against the whole filename (still without the path)?

Reply:
done, see my latest commit.